src/MetricsUploader/MetricsUploaderWindows.cpp

```diff
@@ -56,34 +60,51 @@ ErrorMessageOr<MetricsUploader> MetricsUploader::CreateMetricsUploader(std::stri
     return ErrorMessage("MetricsUploader is already created");
   }
 
+  // Generate UUID as a session identifier.
+  UUID uuid;
+  RPC_STATUS create_status = UuidCreate(&uuid);
+  if (create_status != RPC_S_OK) {
+    return ErrorMessage("Unable to create UUID for metrics uploader");
+  }
+  RPC_CSTR uuid_c_str = nullptr;
+  RPC_STATUS convert_status = UuidToStringA(&uuid, &uuid_c_str);
+  if (convert_status != RPC_S_OK) {
+    return ErrorMessage("Unable to convert UUID to string for metrics uploader");
+  }
+
+  std::string session_uuid{reinterpret_cast<char*>(uuid_c_str)};
+  RpcStringFreeA(&uuid_c_str);
+  LOG("Session UUID for metrics: %s", session_uuid);
+
```

Comment:
Maybe we can move it to the separate function?

Reply:
I'm also not sure that we should fail if the error occurred during session_id generation. I think most of the metrics would still make sense. @antonrohr  @beckerhe WDYT?

Reply:
In general I agree. But UUID generation doesn't fail nowadays. This API function is very old and was designed when it was still common to deduce the UUID from a bit of randomness and the MAC address.

So I guess we could also just fail when UUID generation fails and simplify our code that way. It will not happen in practice.

Reply:
I moved it into a seperate function and added a quick test case. ptal